analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpExecutableLinesMetric.cs

```diff
@@ -132,5 +145,29 @@ private bool HasExcludedCodeAttribute(SyntaxNode node, SyntaxList<AttributeListS
             private bool IsExcludedAttribute(AttributeSyntax attribute) =>
                 attribute.IsKnownType(KnownType.System_Diagnostics_CodeAnalysis_ExcludeFromCodeCoverageAttribute, model);
         }
+
+        private sealed class RazorExecutableLinesWalker : ExecutableLinesWalker
+        {
+            public RazorExecutableLinesWalker(SemanticModel model) : base(model) { }
+
+            protected override bool AddExecutableLineNumbers(Location location)
+            {
+                var mappedLocation = location.GetMappedLineSpan();
```

Comment:
There are cases when, in a razor-generated file we have the content from other files as well. E.g. the content from `_Imports.razor`. 

e.g.
```
#pragma checksum "C:\src\work\Razor.Analyzers\Razor.Analyzers.Mvc.Net7.Example\Pages\Event.razor" "{8829d00f-11b8-4213-878b-770e8597ac16}" "11a9e367f87bf1ac05e400982758ce25819bf64a64d84ac08bea73a203cae8ba"

...

#line 2 "C:\src\work\Razor.Analyzers\Razor.Analyzers.Mvc.Net7.Example\_Imports.razor"
using Microsoft.AspNetCore.Authorization;
#line 8 "C:\src\work\Razor.Analyzers\Razor.Analyzers.Mvc.Net7.Example\Pages\Event.razor"
                      newHeading

```

In this case, the mapped location might point to a file that is not the one we want to compute the metric for. I do not have an example in mind where executable code could be imported (using are not considered executable code) but this doesn't mean that it could not happen. 

What do you think? Should we check the path of the mapped location or it's not actually needed?